File: Problems/Algorithms/1337. The K Weakest Rows in a Matrix/k_weakest_rows.py
class Solution:
    def kWeakestRows(self, mat: List[List[int]], k: int) -> List[int]:
        counts = []
        result = []

        for i in range(len(mat)):
            counts.append([i, sum(mat[i])])

        return [sorted(counts, key=lambda x: x[1])[i][0] for i in range(k)] 

File: Problems/Algorithms/1337. The K Weakest Rows in a Matrix/KWeakestRows.kt
class Solution {
    fun kWeakestRows(mat: Array<IntArray>, k: Int): IntArray {
        val map = mutableMapOf<Int, Int>()
        val result = mutableListOf<Int>()
    
        for (i in mat.indices) {
            val count = mat[i].count { it == 1 }
            map[i] = count
        }
        
        map.toList().sortedBy { (_, value) -> value }.forEach { result.add(it.first) } 
        return result.take(k).toIntArray()
    }
}

